test/integration-migrated/istio/traces_istio_routing_test.go

```diff
@@ -0,0 +1,203 @@
+	verifyAppIsRunning(t, appNs, map[string]string{"app.kubernetes.io/name": "metric-producer"})
+	assert.DeploymentReady(t, kitkyma.TraceGatewayName)
+	assert.TracePipelineHealthy(t, pipeline1Name)
+	assert.TracePipelineHealthy(t, pipeline2Name)
+
+	Eventually(func(g Gomega) {
+		resp, err := suite.ProxyClient.Get(metricServiceURL)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+		defer resp.Body.Close()
+	}, periodic.EventuallyTimeout, periodic.DefaultInterval).Should(Succeed())
+
+	for _, podURLs := range []string{appURL, istiofiedAppURL} {
+		Eventually(func(g Gomega) {
+			resp, err := suite.ProxyClient.Get(podURLs)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+			defer resp.Body.Close()
+		}, periodic.EventuallyTimeout, periodic.DefaultInterval).Should(Succeed())
+	}
+
+	verifyIstioSpans(t, backendExportURL, istiofiedAppNs)
+	verifyIstioSpans(t, istiofiedBackendExportURL, istiofiedAppNs)
+	verifyCustomAppSpans(t, backendExportURL, istiofiedAppName, istiofiedAppNs)
+	verifyCustomAppSpans(t, istiofiedBackendExportURL, istiofiedAppName, istiofiedAppNs)
+	verifyCustomAppSpans(t, backendExportURL, appName, appNs)
+	verifyCustomAppSpans(t, istiofiedBackendExportURL, appName, appNs)
+	verifyNoIstioNoiseSpans(t, backendExportURL)
+	verifyNoIstioNoiseSpans(t, istiofiedBackendExportURL)
+}
+
+func verifySidecarPresent(t testkit.T, namespace string, labelSelector map[string]string) {
+	t.Helper()
+
+	Eventually(func(g Gomega) {
+		listOptions := client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labelSelector),
+			Namespace:     namespace,
+		}
+
+		hasIstioSidecar, err := assert.PodsHaveContainer(t, listOptions, "istio-proxy")
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(hasIstioSidecar).To(BeTrueBecause("Istio sidecar not present"))
+	}, periodic.EventuallyTimeout*2, periodic.DefaultInterval).Should(Succeed())
+}
+
+func verifyAppIsRunning(t testkit.T, namespace string, labelSelector map[string]string) {
+	t.Helper()
+
+	listOptions := client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labelSelector),
+		Namespace:     namespace,
+	}
+
+	assert.PodsReady(t, listOptions)
+}
+
+func verifyIstioSpans(t testkit.T, backendURL, namespace string) {
+	t.Helper()
+
+	Eventually(func(g Gomega) {
+		resp, err := suite.ProxyClient.Get(backendURL)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+		defer resp.Body.Close()
+
+		g.Expect(resp).To(HaveHTTPBody(HaveFlatTraces(ContainElement(SatisfyAll(
+			// Identify istio-proxy traces by component=proxy attribute
+			HaveSpanAttributes(HaveKeyWithValue("component", "proxy")),
+			HaveSpanAttributes(HaveKeyWithValue("istio.namespace", namespace)),
+		)))))
+	}, periodic.TelemetryEventuallyTimeout, periodic.TelemetryInterval).Should(Succeed())
+}
+
+func verifyNoIstioNoiseSpans(t testkit.T, backendURL string) {
```

Comment:
can we use assert instead of verify ? Also please check if we can use internal libraries and may be we dont need them any more now?